### PR TITLE
fix(parquet): support read i96 timestamp from parquet file

### LINKED ISSUE #6627: bug:  load parquet data to databend failed  if had timestamp  columns
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

80e09b30c1d5bbdcfdc244c997a10374fd797257

### What's Wrong?

could't load parquet data to databend

### How to Reproduce?

1. create table with sql "create table mytime(id string, t_bool boolean, t_float float, t_double double,  t_timestamp timestamp, t_data date, t_array array(int));"
2. load parquet data,    curl -XPUT 'http://root:@127.0.0.1:8081/v1/streaming_load' -H 'insert_sql: insert into mytime format Parquet' -F 'upload=@"./mytime.parquet"'

parquet data:  https://github.com/sandflee/databend/blob/parquet/mytime.parquet

 

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

It's compatible change:

```
create table t( a timestamp(0), b timestamp(3), d timestamp(6));
insert into t select now(), now(), now() from numbers(10);

create table t2( a timestamp(0), b timestamp(3), d timestamp(6));
insert into t2 select now(), now(), now() from numbers(10);
```

`databend-query` with Old/New version can read table t and t2.

Fixes #6627

### CODE REVIEW

File: common/datavalues/src/types/type_timestamp.rs [hunk 1]
```diff
@@ -141,7 +142,7 @@ impl DataType for TimestampType {
     }
 
     fn arrow_type(&self) -> ArrowType {
-        ArrowType::Int64
+        ArrowType::Timestamp(TimeUnit::Microsecond, None)
     }
 
     fn custom_arrow_meta(&self) -> Option<BTreeMap<String, String>> {
```
Comment (b41sh) [inline common/datavalues/src/types/type_timestamp.rs:148]: we can remove `custom_arrow_meta`

File: common/datavalues/src/types/data_type.rs [hunk 2]
```diff
@@ -185,7 +185,7 @@ pub fn from_arrow_type(dt: &ArrowType) -> DataTypeImpl {
             DataTypeImpl::String(StringType::default())
         }
 
-        ArrowType::Timestamp(_, _) => DataTypeImpl::Timestamp(TimestampType::create(0)),
+        ArrowType::Timestamp(_, _) => DataTypeImpl::Timestamp(TimestampType::create(3)),
```
Comment (b41sh) [inline common/datavalues/src/types/data_type.rs:188]: `precision` can be generated based on `TimeUnit`

File: common/datavalues/src/types/type_timestamp.rs (hunk 1 above)
Comment (sundy-li) [reply common/datavalues/src/types/type_timestamp.rs:148]: We still need `custom_arrow_meta ` to keep compatibility with old version meta.

Comment (b41sh) [reply common/datavalues/src/types/type_timestamp.rs:148]: ok

**Comment by dantengsky** [comment]
- confirm that data(with timestamp column) generated by the old version, could be read by this PR


test_stateless_standalone_linux failed.

[test_stateless_standalone_linux](https://github.com/datafuselabs/databend/runs/7382929173?check_suite_focus=true#logs)


> 
> 03_0023_insert_into_array:                                              [ FAIL ] - result differs with:
> --- /workspace/tests/suites/0_stateless/03_dml/03_0023_insert_into_array.result	2022-07-18 04:15:39.394586214 +0000
> +++ /workspace/tests/suites/0_stateless/03_dml/03_0023_insert_into_array.stdout	2022-07-18 04:19:11.488000857 +0000
> @@ -1,3 +1,5 @@
> +ERROR 1105 (HY000) at line 118: Code: 1104, displayText = called `Result::unwrap()` on an `Err` value: OutOfSpec("ListArray's child's DataType must match. However, the expected DataType is Timestamp(Microsecond, None) while it got Int64.").
> +ERROR 1105 (HY000) at line 119: Code: 1054, displayText = Expected server error code: 1010 but got: 1002.
>  ==Array(UInt8)==
>  1	[1, 2, 3]
>  2	[254, 255]
> @@ -63,10 +65,6 @@
>  2021-01-01	2022-01-01
>  1990-12-01	2030-01-12
>  ==Array(Timestamp)==
> -1	['2021-01-01 01:01:01.000000', '2022-01-01 01:01:01.000000']
> -2	['1990-12-01 10:11:12.000000', '2030-01-12 22:00:00.000000']
> -2021-01-01 01:01:01.000000	2022-01-01 01:01:01.000000
> -1990-12-01 10:11:12.000000	2030-01-12 22:00:00.000000
>  ==Array(String)==
>  1	['aa', 'bb']
>  2	['cc', 'dd']





> 03_0023_insert_into_array_v2:                                           [ FAIL ] - result differs with:
> --- /workspace/tests/suites/0_stateless/03_dml/03_0023_insert_into_array_v2.result	2022-07-18 04:15:39.394586214 +0000
> +++ /workspace/tests/suites/0_stateless/03_dml/03_0023_insert_into_array_v2.stdout	2022-07-18 04:19:15.588107257 +0000
> @@ -1,3 +1,5 @@
> +ERROR 1105 (HY000) at line 118: Code: 1104, displayText = called `Result::unwrap()` on an `Err` value: OutOfSpec("ListArray's child's DataType must match. However, the expected DataType is Timestamp(Microsecond, None) while it got Int64.").
> +ERROR 1105 (HY000) at line 119: Code: 1054, displayText = Expected server error code: 1010 but got: 1002.
>  ==Array(UInt8)==
>  1	[1, 2, 3]
>  2	[254, 255]
> @@ -63,10 +65,6 @@
>  2021-01-01	2022-01-01
>  1990-12-01	2030-01-12
>  ==Array(Timestamp)==
> -1	['2021-01-01 01:01:01.000000', '2022-01-01 01:01:01.000000']
> -2	['1990-12-01 10:11:12.000000', '2030-01-12 22:00:00.000000']
> -2021-01-01 01:01:01.000000	2022-01-01 01:01:01.000000
> -1990-12-01 10:11:12.000000	2030-01-12 22:00:00.000000
>  ==Array(String)==
>  1	['aa', 'bb']
>  2	['cc', 'dd']
>

**Comment by BohuTANG** [comment]
> * confirm that data(with timestamp column) generated by the old version, could be read by this PR

I think this issue needs to be a high priority https://github.com/datafuselabs/databend/issues/6557
Now it's a difficult for us to find the query layer compatibility issue, like this PR we need a strong test for it.

**Comment by dantengsky** [comment]
> I think this issue needs to be a high priority #6557 Now it's a difficult for us to find the query layer compatibility issue, like this PR we need a strong test for it.

Agree. #6557 will be addressed just after #6639 closed

**Review by dantengsky — COMMENTED** [review]
LGTM

**Review by dantengsky — APPROVED** [review]
LGTM

**Review by BohuTANG — APPROVED** [review]

**Comment by b41sh** [comment]
LGTM

**Pull request merged**